annotatedcsv/reader.go

```diff
@@ -0,0 +1,342 @@
+// Copyright 2021 InfluxData, Inc. All rights reserved.
+// Use of this source code is governed by MIT
+// license that can be found in the LICENSE file.
+
+// Package annotatedcsv provides a reader for annotated CSV sources.
+// Annotated CSV must contain at least an annotation with data types definition.
```

Comment:
How about linking to the description of the format here?

Reply:
Do you mean linking to Flux spec? Yes, but we should keep a note that the reader needs at least the `dataType` annotation.

Reply:
Yeah, I'm thinking of including these links:

https://docs.influxdata.com/influxdb/cloud/reference/syntax/annotated-csv/
https://docs.influxdata.com/influxdb/cloud/reference/syntax/annotated-csv/extended/
https://www.w3.org/TR/tabular-data-model/

Reply:
Interesting, I haven't seen annotated CSV reference before. My source of CSV format description was the Flux spec.
The reader must be a bit improved to handle few more things, like `dateTime:number`.

However, some stuff from extended annotated CSV reference is not implemented yet (extended annotations,  extended type specification). As it is not supported in Flux result now, I guess we can skip it for now.

So, I would just add a reference to https://docs.influxdata.com/influxdb/cloud/reference/syntax/annotated-csv/. 
And, as it contains a CSV example I will remove the CSV example from the package doc.

As the reference describes tables, I'm for returning back to the term __table_ for the current __section__. WDYT?

Reply:
I think we should stick with "section" as that's been okayed with the flux team, and avoids the ambiguity with the "table" column.